storagenode/piecestore/verification.go

```diff
@@ -149,3 +149,27 @@ func (endpoint *Endpoint) IsExpired(expiration *timestamp.Timestamp) bool {
 	// TODO: return specific error about either exceeding the expiration completely or just the grace period
 	return expirationTime.Before(time.Now().Add(-endpoint.config.ExpirationGracePeriod))
 }
+
+// VerifyAvailableSpace verifies whether there is available disk space
+func (endpoint *Endpoint) VerifyAvailableSpace(ctx context.Context, limit *pb.OrderLimit2) error {
+	availableSpace, err := endpoint.monitor.AvailableSpace(ctx)
+	if err != nil {
+		return ErrInternal.Wrap(err)
+	}
+	if availableSpace < limit.Limit {
```

Comment:
If the order limit is significantly higher than the actual amount being downloaded/uploaded then the storage node could be penalized for rejecting even though it might have sufficient bandwidth or storage space

Reply:
previously we would verify available bandwidth/disk on every read/write. Why are we not doing that anymore?

Reply:
there's a todo note somewhere about it, but yeah, we should verfiy and update it on each action

Reply:
this seems like the pr to put it back

Reply:
so be it

Reply:
The logic should roughly be:

```
endpoint.monitor.AddBandwith(ctx, limit.Amount)
defer endpoint.monitor.AddBandwith(ctx, order.Amount - limit.Amount)
```

Reply:
I'm not sure if im following, we need to check space/bandwidth on each read/write. As @aleitner  wrote we cannot use OrderLimit.Limit because it can be much higher than real data size for download/upload.

Reply:
requests usually come in with limits with the maximum they intend to read, which is either "size of the piece" or "a specific size in case of audits"... generally the difference shouldn't be that high... Of course we can do it incrementally as well.